src/interfaces/interfaces.ts

```diff
@@ -176,6 +185,14 @@ namespace interfaces {
         getAll<T>(serviceIdentifier: ServiceIdentifier<T>): T[];
         getAllTagged<T>(serviceIdentifier: ServiceIdentifier<T>, key: string | number | symbol, value: any): T[];
         getAllNamed<T>(serviceIdentifier: ServiceIdentifier<T>, named: string | number | symbol): T[];
+        getAsync<T>(serviceIdentifier: ServiceIdentifier<T>): Promise<T>;
+        getNamedAsync<T>(serviceIdentifier: ServiceIdentifier<T>, named: string | number | symbol): Promise<T>;
+        getTaggedAsync<T>(serviceIdentifier: ServiceIdentifier<T>, key: string | number | symbol, value: any): Promise<T>;
```

Comment:
we should really figure out a way to do a generic for this value and then apply it to all this kinds of methods.
one suggestion will be to either make it as an unknown or figure out all the types that might come into this and just  type it as a union of all the types

Reply:
I think value can receive any type, so maybe `unknown` is the best type to apply